include/marisa/agent.h

```diff
@@ -36,6 +46,16 @@ class Agent {
     return *state_;
   }
 
+#if __cplusplus >= 201703L
+  void set_key(const std::string_view str) {
```

Comment:
nit: Remove `const`. It is an implementation detail that `string_view` isn't modified inside the method.

Reply:
I'm not sure what you have in mind. Maybe:
```
// implementation detail, does nothing here
void f(const T x);
// not implementation detail, does something
void f(const T x) { ... }
```

This can prevent bugs (unlikely in this case since the function is two lines).

In any case, I've removed the const since `marisa-trie` style doesn't like const on args, even in definitions.